### PR TITLE
Add .textlintignore support

### LINKED ISSUE #500: Support .textlintignore
Currently, textlint allows specifying files by allowing list.
In other words, textlint disallows to ignore some files by deny list.

https://github.com/textlint/textlint/blob/442c297c60e77de55f07fea7c75d71ac05a6a03a/packages/textlint/src/util/find-util.ts#L35-L60

We want to support ignoring via `.textlintignore`.
The user can tell textlint to ignore specific files and directories by creating an `.textlintignore` file.

## Task

- [ ] Support `.textlintignore`
- [ ] Support `--ignore-path` as CLI option
  - Path to a file containing patterns that describe files to ignore.
  -  By default, `./.textlintignore`

## Related

- [Configuring ESLint - ESLint - Pluggable JavaScript linter](https://eslint.org/docs/user-guide/configuring)
- [CLI · Prettier](https://prettier.io/docs/en/cli.html#ignore-path)

### PR DESCRIPTION
**Pull request by frozenbonito**
close #500.

This PR contains a trivial breaking change, `node_modules` and `.git` will be ignored by default.

### CODE REVIEW

File: packages/textlint/test/engine/textlint-engine-test.ts [hunk 1]
```diff
@@ -290,6 +290,23 @@ describe("textlint-engine-test", function () {
...
+            it("should ignore file described in the ignore file", function () {
+                const ignoreFilePath = path.join(__dirname, "fixtures/.textlintignore");
+                const engine = new TextLintEngine({ rulePaths: [rulesDir], ignoreFile: ignoreFilePath });
+                const filePath = path.join(__dirname, "fixtures/*.md");
+                return engine.executeOnFiles([filePath]).then((results) => {
+                    assert.ok(Array.isArray(results));
+                    assert.ok(results.length > 0);
+                    assert.ok(
+                        results.every((result) => {
+                            console.log(result.filePath);
```
Comment (azu) [inline packages/textlint/test/engine/textlint-engine-test.ts:303]: Can you remove it?

Comment (azu) [inline packages/textlint/test/engine/textlint-engine-test.ts:303]:
```suggestion
```

File: packages/textlint/src/util/find-util.ts [hunk 2]
```diff
@@ -46,11 +76,14 @@ export function findFiles(patterns: string[], options: { cwd?: string } = {}): s
...
-                nodir: true
+                cwd,
+                absolute: true,
+                nodir: true,
+                ignore: ignoredPatterns
             }).forEach((filePath: string) => {
                 // workaround for windows
                 // https://github.com/isaacs/node-glob/issues/74#issuecomment-31548810
-                addFile(path.resolve(filePath));
+                addFile(path.resolve(cwd, filePath));
```
Comment (azu) [inline packages/textlint/src/util/find-util.ts:86]: [It is just a question. not need to change]

Is this `cwd` needed?
`absolute` option makes `filePath` to absolute path.
So, I think that `cwd` is always just ignored.

**Review by azu — APPROVED** [review]
Thanks!
LGTM.

I've commented on a little thing.

**Comment by azu** [comment]
📝 Also, we need to add a document like "Ignoring Text" to https://github.com/textlint/textlint/tree/master/docs

It will be similar one of https://prettier.io/docs/en/ignore.html

File: packages/textlint/src/util/find-util.ts (hunk 2 above)
Comment (frozenbonito) [reply packages/textlint/src/util/find-util.ts]: You are quite right, `cwd` is no longer needed.
I removed it.

File: packages/textlint/test/engine/textlint-engine-test.ts (hunk 1 above)
Comment (frozenbonito) [reply packages/textlint/test/engine/textlint-engine-test.ts]: Sorry, it's my mistake.
I removed it.

**Comment by frozenbonito** [comment]
I added "Ignore Text" document.

**Comment by azu** [comment]
https://deploy-preview-748--textlint.netlify.app/docs/ignore.html
Looks Nice!

**Comment by azu** [comment]
@frozenbonito Thanks!
I've released https://github.com/textlint/textlint/releases/tag/textlint%4011.9.0

**Pull request merged**